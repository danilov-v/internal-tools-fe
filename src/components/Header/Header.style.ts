import styled from 'styled-components';
import { Link } from '@reach/router';

type HumburgerProps = {
  expanded: boolean;
};

export const Header = styled.header`
  position: fixed;
  top: 0;
  left: 4vw;
  z-index: 100;
  width: 92vw;
  min-height: 65px;
  padding-top: 1em;
  font-size: 1.2rem;
  border-bottom: 2px solid #616262ff;

  @media screen and (min-width: 768px) {
    display: flex;
    padding-top: 0;
    align-items: flex-end;
    padding-bottom: 1em;
  }
`;

export const NavBar = styled.ul<HumburgerProps>`
  list-style-type: none;
  padding: 0;
  display: ${({ expanded }) => (expanded ? 'block' : 'none')};
  margin-top: 3rem;

  @media screen and (min-width: 768px) {
    margin: 0;
    display: flex;
    flex-direction: row;
    justify-content: flex-start;
    align-items: baseline;
  }
`;

export const NavItem = styled.li`
  text-align: center;
  margin-left: 2.2rem;

  @media screen and (max-width: 768px) {
    margin: 0;

    &:not(:first-child) {
      margin-top: 1.2rem;
    }
  }
`;

export const NavLink = styled(Link)`
  color: #cccccc;
  text-decoration: none;
  position: relative;

  &:hover {
    color: #ffdc6282;
  }
  &[state='active'] {
    color: #ffdc62ff;

    &::before {
      content: '';
      display: block;
      position: absolute;
      top: -1rem;
      left: 40%;
      width: 12px;
      height: 12px;
      background-color: #ffdc62ff;
      clip-path: polygon(50% 100%, 0 0, 100% 0);
    }
  }
`;

export const Humburger = styled.div<HumburgerProps>`
  position: absolute;
  top: 1em;
  right: 0;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  width: 3rem;
  height: 3rem;
  transform: rotate(0deg);
  transition: 0.5s ease-in-out;
  cursor: pointer;

  @media screen and (min-width: 768px) {
    display: none;
  }

  div {
    height: 0.25rem;
    background: #cccccc;
    border-radius: 9px;
    opacity: 1;
    left: 0;
    transform-origin: left center;
    transition: 0.25s ease-in-out;
  }

  & div:nth-child(1) {
    transform: ${({ expanded }) =>
      expanded ? 'rotate(45deg)' : 'rotate(0deg)'};
  }
  & div:nth-child(2) {
    opacity: ${({ expanded }) => (expanded ? 0 : 1)};
  }
  & div:nth-child(3) {
    transform: ${({ expanded }) =>
      expanded ? 'rotate(-45deg)' : 'rotate(0deg)'};
  }
`;

export const UserAvatar = styled.div`
  position: absolute;
  top: 1.6em;
  right: calc(100vw - 260px);

  @media screen and (min-width: 768px) {
    right: 0;
  }
`;
