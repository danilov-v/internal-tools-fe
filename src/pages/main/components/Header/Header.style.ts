import styled from 'styled-components';
import { Link } from '@reach/router';

type HumburgerProps = {
  expanded: boolean;
};

const LINE_COLOR = '#616262';

export const Header = styled.header`
  position: sticky;
  top: 0;
  display: flex;
  justify-content: space-between;
  padding: 15px 0;
  z-index: 100;
  border-bottom: 2px solid ${LINE_COLOR};
  background: linear-gradient(90deg, #44494b 0%, #4c5153 25%, #4c5153 100%);

  @media screen and (min-width: 768px) {
    align-items: flex-end;
    justify-content: initial;
  }
`;

export const Logo = styled.div`
  cursor: pointer;
`;

export const NavBar = styled.div<HumburgerProps>`
  list-style-type: none;
  padding: 0;
  display: ${({ expanded }) => (expanded ? 'flex' : 'none')};
  flex-direction: column;
  margin-top: 80px;

  @media screen and (min-width: 768px) {
    display: block;
    flex: 1;
    margin-top: 0;
  }
`;

export const NavItem = styled.div`
  display: block;
  text-align: center;
  margin-bottom: 20px;

  @media screen and (min-width: 768px) {
    display: inline-block;
    margin-left: 40px;
    margin-bottom: 0;
  }
`;

export const NavLink = styled(Link)`
  color: #cccccc;
  text-decoration: none;
  position: relative;

  &:hover {
    color: #ffdc6282;
  }
  &[state='active'] {
    color: #ffdc62;

    &::before {
      content: '';
      display: block;
      position: absolute;
      top: -12px;
      left: calc(50% - 5px);
      width: 10px;
      height: 8px;
      background-color: #ffdc62;
      clip-path: polygon(50% 100%, 0 0, 100% 0);
    }
  }
`;

export const Humburger = styled.div<HumburgerProps>`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  width: 45px;
  height: 35px;
  transform: rotate(0deg);
  transition: 0.5s ease-in-out;
  cursor: pointer;

  @media screen and (min-width: 768px) {
    display: none;
  }

  div {
    height: 3px;
    background: #cccccc;
    border-radius: 2px;
    opacity: 1;
    left: 0;
    transform-origin: left center;
    transition: 0.25s ease-in-out;
  }

  & div:nth-child(1) {
    transform: ${({ expanded }) =>
      expanded ? 'rotate(45deg)' : 'rotate(0deg)'};
  }
  & div:nth-child(2) {
    opacity: ${({ expanded }) => (expanded ? 0 : 1)};
  }
  & div:nth-child(3) {
    transform: ${({ expanded }) =>
      expanded ? 'rotate(-45deg)' : 'rotate(0deg)'};
  }
`;

export const UserRank = styled.div`
  display: none;

  @media screen and (min-width: 768px) {
    display: block;
  }
`;

export const Actions = styled.div`
  display: flex;
  align-items: flex-end;
  border-left: 2px solid ${LINE_COLOR};
  margin-left: 20px;
  height: 40px;
  justify-content: space-around;
  flex-basis: 120px;
`;

export const Action = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  flex-basis: 30px;
  padding: 4px 0;
  cursor: pointer;

  &:hover {
    opacity: 0.7;
  }
`;

export const ActionIcon = styled.img`
  width: 15px;
  height: 15px;
`;
