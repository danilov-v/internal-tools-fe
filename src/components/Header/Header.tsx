import React from 'react';
import { useLocalStore, useObserver } from 'mobx-react-lite';

import * as S from './Header.style';

type HeaderProps = {
  test: string;
};

const Header: React.FC<HeaderProps> = ({ test }) => {
  const counterStore = useLocalStore(() => ({
    counter: 0,
    increment() {
      counterStore.counter += 1;
    },
  }));

  return useObserver(() => (
    <S.Header>
      There will be header. <br />
      Test params - {test}! <br />
      Test store value - {counterStore.counter} <br />
      <button type="button" onClick={counterStore.increment}>
        Increment
      </button>
    </S.Header>
  ));
};

export { Header };
