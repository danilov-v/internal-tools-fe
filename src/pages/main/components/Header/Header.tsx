import React, { useState } from 'react';

import { Logo } from 'components/Logo';
import { Avatar } from 'pages/main/components/Avatar';
import * as S from './Header.style';

type IsActivePropsType = {
  isPartiallyCurrent: boolean;
};

const isActive = ({ isPartiallyCurrent }: IsActivePropsType): object => {
  return { state: isPartiallyCurrent ? 'active' : 'not-active' };
};

const testUser = {
  lastName: 'Стороженко',
  rank: 'captain',
  position: 'Командир роты',
};

const Header: React.FC<{}> = () => {
  const [isExpanded, setExpanded] = useState(false);
  const toggleExpanded = (): void => setExpanded(!isExpanded);

  return (
    <S.Header>
      <S.UserLogo>
        <Logo />
      </S.UserLogo>
      <S.NavBar expanded={isExpanded}>
        <S.NavItem>
          <S.NavLink getProps={isActive} to="/personnel">
            Рота
          </S.NavLink>
        </S.NavItem>
        <S.NavItem>
          <S.NavLink getProps={isActive} to="/schedule">
            График
          </S.NavLink>
        </S.NavItem>
      </S.NavBar>
      <S.UserAvatar>
        <Avatar user={testUser} />
      </S.UserAvatar>
      <S.Humburger expanded={isExpanded} onClick={toggleExpanded}>
        <div />
        <div />
        <div />
      </S.Humburger>
    </S.Header>
  );
};

export { Header };
