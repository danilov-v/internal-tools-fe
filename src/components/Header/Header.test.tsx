import React from 'react';
import { render, RenderResult } from '@testing-library/react';
import 'jest-styled-components';

import { Header } from './Header';

describe('Header component', () => {
  const getComponent = (): RenderResult => {
    return render(<Header />);
  };

  it('matches snapshot', () => {
    expect(getComponent().baseElement).toMatchSnapshot();
  });
});
