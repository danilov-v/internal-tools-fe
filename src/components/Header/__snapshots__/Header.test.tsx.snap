// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component matches snapshot 1`] = `
.c1 {
  height: 3rem;
  width: 3rem;
  width: 3rem;
  margin-bottom: 0px;
  display: block;
}

.c2 {
  display: block;
  width: 100%;
  height: 100%;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
}

.c8 {
  width: 2rem;
  height: 2rem;
  display: inline-block;
  max-width: 100%;
}

.c9 {
  font-size: 1.2rem;
  color: #ccccccff;
  margin-left: 5px;
}

.c9 span {
  position: absolute;
  bottom: 1.6rem;
  font-size: 0.8rem;
  color: #727374ff;
}

.c0 {
  position: fixed;
  top: 0;
  left: 4vw;
  z-index: 100;
  width: 92vw;
  min-height: 65px;
  padding-top: 1em;
  font-size: 1.2rem;
  border-bottom: 2px solid #616262ff;
}

.c3 {
  list-style-type: none;
  padding: 0;
  display: none;
  margin-top: 3rem;
}

.c4 {
  text-align: center;
  margin-left: 2.2rem;
}

.c5 {
  color: #ccccccff;
  -webkit-text-decoration: none;
  text-decoration: none;
  position: relative;
}

.c5:hover {
  color: #ffdc6282;
}

.c5[state='active'] {
  color: #ffdc62ff;
}

.c5[state='active']::before {
  content: '';
  display: block;
  position: absolute;
  top: -1rem;
  left: 40%;
  width: 12px;
  height: 12px;
  background-color: #ffdc62ff;
  -webkit-clip-path: polygon(50% 100%,0 0,100% 0);
  clip-path: polygon(50% 100%,0 0,100% 0);
}

.c10 {
  position: absolute;
  top: 1em;
  right: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  width: 3rem;
  height: 3rem;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  -webkit-transition: 0.5s ease-in-out;
  transition: 0.5s ease-in-out;
  cursor: pointer;
}

.c10 div {
  height: 0.25rem;
  background: #ccccccff;
  border-radius: 9px;
  opacity: 1;
  left: 0;
  -webkit-transform-origin: left center;
  -ms-transform-origin: left center;
  transform-origin: left center;
  -webkit-transition: 0.25s ease-in-out;
  transition: 0.25s ease-in-out;
}

.c10 div:nth-child(1) {
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
}

.c10 div:nth-child(2) {
  opacity: 1;
}

.c10 div:nth-child(3) {
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
}

.c6 {
  position: absolute;
  top: 1.6em;
  right: calc(100vw - 260px);
}

@media screen and (min-width:768px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    padding-top: 0;
    -webkit-align-items: flex-end;
    -webkit-box-align: flex-end;
    -ms-flex-align: flex-end;
    align-items: flex-end;
    padding-bottom: 1em;
  }
}

@media screen and (min-width:768px) {
  .c3 {
    margin: 0;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
    -webkit-align-items: baseline;
    -webkit-box-align: baseline;
    -ms-flex-align: baseline;
    align-items: baseline;
  }
}

@media screen and (max-width:768px) {
  .c4 {
    margin: 0;
  }

  .c4:not(:first-child) {
    margin-top: 1.2rem;
  }
}

@media screen and (min-width:768px) {
  .c10 {
    display: none;
  }
}

@media screen and (min-width:768px) {
  .c6 {
    right: 0;
  }
}

<body>
  <div>
    <header
      class="c0"
    >
      <div
        class="c1"
      >
        <img
          alt="logo"
          class="c2"
          src="logo.svg"
        />
      </div>
      <ul
        class="c3"
      >
        <li
          class="c4"
        >
          <a
            class="c5"
            href="/main"
            state="not-active"
          >
            Рота
          </a>
        </li>
        <li
          class="c4"
        >
          <a
            class="c5"
            href="/#main"
            state="not-active"
          >
            График
          </a>
        </li>
      </ul>
      <div
        class="c6"
      >
        <div
          class="c7"
        >
          <img
            alt="Avatar"
            class="c8"
            src="captain_rang.svg"
          />
          <div
            class="c9"
          >
            <span>
              Командир роты
            </span>
            Стороженко
          </div>
        </div>
      </div>
      <div
        class="c10"
      >
        <div />
        <div />
        <div />
      </div>
    </header>
  </div>
</body>
`;
